src/Image/Darkroom/ImageMagick.php

```diff
@@ -72,8 +72,47 @@ protected function convert(string $file, array $options): string
 			}
 		}
 
+		// frame option to allow selecting layers for multi-layer or frames for animated images
+		$fileOptions = '';
+		$frame = $options['frame'];
+		$maxFrames = $this->frameCount($file, $options);
+		if ($frame !== null) {
+			// check whether frame is in bounds
+			if ($frame < 0) {
+				throw new Exception('Frame option must be a positive integer');
+			}
+
+			if ($frame !== 0 && $frame >= $maxFrames) {
+				throw new Exception('Frame option must be smaller than the number of frames in the image');
+			}
+
+			$fileOptions = "[{$frame}]";
+		} elseif ($maxFrames > 1) {
+			// if frame is not set and target format doesn't support multi-layer images, select first frame
+			$targetFormat = $options['format'] ?? F::extension($file);
+			$multiLayerFormats = ['gif', 'avif', 'webp'];
+			if (!in_array($targetFormat, $multiLayerFormats)) {
+				$fileOptions = '[0]';
+			}
```

Comment:
I think for better understanding of the code this is twisted: We should first on top check if target format isn't multi frame (why layers?) format and for default set it to first frame, unless a specific frame is defined.

Reply:
I named it layers, probably because different formats handle frames differently. Some are animated (like GIFs), some have them as separate layers for e.g. graphics editing (e.g. TIFFs). Although you're right that in web behavior it's probably only 'multi-frame'.